test/controllers/pensions/UpdatePensionProviderControllerSpec.scala

```diff
@@ -425,12 +363,136 @@ class UpdatePensionProviderControllerSpec extends PlaySpec with FakeTaiPlayAppli
     }
   }
 
+  "redirectUpdatePension" must {
+
+    "redirect to the Do You Get This Pension page when there is no update pension ID cache value present" in {
+      val pensionId = 1
+      val cacheMap = Map(UpdatePensionProvider_IdKey -> pensionId.toString, UpdatePensionProvider_NameKey -> pensionName)
+      val sut = createSUT
+
+      when(taxAccountService.taxCodeIncomes(any(), any())(any())).
+        thenReturn(Future.successful(TaiSuccessResponseWithPayload(Seq(pensionTaxCodeIncome, empTaxCodeIncome))))
+      when(successfulJourneyCacheService.currentValue(Matchers.eq(s"$TrackSuccessfulJourney_UpdatePensionKey-$pensionId"))(any())).
+      thenReturn(Future.successful(None))
+      when(journeyCacheService.cache(Matchers.eq(cacheMap))(any())).thenReturn(Future.successful(cacheMap))
+
+      val result = sut.redirectUpdatePension(pensionId)(RequestBuilder.buildFakeRequestWithAuth("GET"))
+      status(result) mustBe SEE_OTHER
+      redirectLocation(result).get mustBe routes.UpdatePensionProviderController.doYouGetThisPension().url
+    }
+
+    "redirect to the Duplicate Submission Warning page when there is an Update pension ID cache value present" in {
+      val pensionId = 1
+      val cacheMap = Map(UpdatePensionProvider_IdKey -> pensionId.toString, UpdatePensionProvider_NameKey -> pensionName)
+      val sut = createSUT
+
+      when(taxAccountService.taxCodeIncomes(any(), any())(any())).
+        thenReturn(Future.successful(TaiSuccessResponseWithPayload(Seq(pensionTaxCodeIncome, empTaxCodeIncome))))
+      when(successfulJourneyCacheService.currentValue(Matchers.eq(s"$TrackSuccessfulJourney_UpdatePensionKey-$pensionId"))(any())).
+      thenReturn(Future.successful(Some("true")))
+      when(journeyCacheService.cache(Matchers.eq(cacheMap))(any())).thenReturn(Future.successful(cacheMap))
+
+      val result = sut.redirectUpdatePension(pensionId)(RequestBuilder.buildFakeRequestWithAuth("GET"))
+      status(result) mustBe SEE_OTHER
```

Comment:
Dedupe test setup

Reply:
Deduped the fake request, journeyCache Call and taxAccountService Call now. 

Not sure if there's anything else that can be deduped?